Firestore emulator crash on export with ~0.5 GB of data

<!-- DO NOT DELETE
validate_template=true
template_path=.github/ISSUE_TEMPLATE/bug_report.md
-->

<!--
Thank you for contributing to the Firebase community!

Think you found a bug?
=======================
Yeah, we're definitely not perfect! Please use this template and include a minimal repro when opening the issue. If you know how to solve the issue, please create a Pull Request, and we'd be happy to review it!

Have a feature request?
========================
Great, we love hearing how we can improve our products! However, do not use this template to submit a feature request. Please submit your feature requests to: https://firebase.google.com/support/contact/bugs-features/

Have a usage question?
=======================
We get lots of those and we love helping you, but GitHub is not the best place for them and they will be closed. Please take a look at the guide first: https://firebase.google.com/docs/cli/

If the official documentation doesn't help, try asking through our official support channel: https://firebase.google.com/support/

Additional locations to check for solutions or assistance from the community:
- Stack Overflow: https://stackoverflow.com/
- Firebase Slack Community: https://firebase.community/

*Please avoid duplicate posting across multiple channels!*
-->

### [REQUIRED] Environment info

<!-- What version of the Firebase CLI (`firebase-tools`) are you using? Note that your issue may already be fixed in the latest versions. The latest version can be found at https://github.com/firebase/firebase-tools/releases -->

<!-- Output of `firebase --version` -->

**firebase-tools:** 10.2.2

<!-- e.g. macOS, Windows, Ubuntu -->

**Platform:** macOS

### [REQUIRED] Test case

<!-- Provide a minimal, complete, and verifiable example (http://stackoverflow.com/help/mcve) -->
See below

### [REQUIRED] Steps to reproduce

<!-- Provide the steps needed to reproduce the issue with the above test case. -->
Export ~0.56GB of data for the firebase emulator. The emulator crashes and fails to export due to running out of buffer memory. My machine has 64GB ram so this should not be an issue

### [REQUIRED] Expected behavior

<!-- What is the expected behavior? -->
Export completes successfully

### [REQUIRED] Actual behavior

<!-- Run the command with --debug flag, and include the logs below. -->
Export crashes with following error:
```
WARNING: Server fails to send the message, abort the channel!
Mar 18, 2022 12:55:25 PM com.google.cloud.datastore.emulator.firestore.webchannel.FirestoreV1WebChannelAdapter$FirestoreListenHandler onError
WARNING: [ERROR] NETWORK_ERROR ()
Mar 18, 2022 12:55:25 PM com.google.net.webchannel.server.v8.BackChannel add
WARNING: Failed to send a new message due to too many pending messagings in the back channel (10001). May need enable flow control.
Mar 18, 2022 12:55:25 PM com.google.net.webchannel.server.v8.ChannelInternalImpl handle
WARNING: Server fails to send the message, abort the channel!
Mar 18, 2022 12:55:25 PM com.google.cloud.datastore.emulator.firestore.webchannel.FirestoreV1WebChannelAdapter$FirestoreListenHandler onError

<this is repeated for 1.762 million lines>


Mar 18, 2022 1:08:36 PM com.google.cloud.datastore.emulator.impl.util.WrappedStreamObserver onError
INFO: operation failed: null
java.nio.BufferOverflowException
    at java.nio.HeapByteBuffer.put(HeapByteBuffer.java:211)
    at com.google.cloud.datastore.emulator.impl.LevelDBLogWriterChannel.writePhysicalRecord(LevelDBLogWriterChannel.java:154)
    at com.google.cloud.datastore.emulator.impl.LevelDBLogWriterChannel.write(LevelDBLogWriterChannel.java:77)
    at com.google.cloud.datastore.emulator.impl.LevelDBLogWriterChannel.write(LevelDBLogWriterChannel.java:56)
    at com.google.cloud.datastore.emulator.impl.ExportImportUtil.writeEntitiesToFile(ExportImportUtil.java:182)
    at com.google.cloud.datastore.emulator.impl.ExportImportUtil.exportEntitiesForSpec(ExportImportUtil.java:157)
    at com.google.cloud.datastore.emulator.impl.ExportImportUtil.exportEntities(ExportImportUtil.java:89)
    at com.google.cloud.datastore.emulator.impl.CloudFirestoreV1.exportData(CloudFirestoreV1.java:688)
    at com.google.cloud.datastore.emulator.impl.CloudFirestoreV1Router.exportData(CloudFirestoreV1Router.java:206)
    at com.google.cloud.datastore.emulator.firestore.EmulatorAuxiliaryGrpcAdapter$1.lambda$exportData$3(EmulatorAuxiliaryGrpcAdapter.java:98)
    at com.google.cloud.datastore.emulator.firestore.EmulatorAuxiliaryGrpcAdapter.unary(EmulatorAuxiliaryGrpcAdapter.java:49)
    at com.google.cloud.datastore.emulator.firestore.EmulatorAuxiliaryGrpcAdapter.access$000(EmulatorAuxiliaryGrpcAdapter.java:28)
    at com.google.cloud.datastore.emulator.firestore.EmulatorAuxiliaryGrpcAdapter$1.exportData(EmulatorAuxiliaryGrpcAdapter.java:95)
    at com.google.firestore.emulator.v1.FirestoreEmulatorGrpc$MethodHandlers.invoke(FirestoreEmulatorGrpc.java:631)
    at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:182)
    at io.grpc.PartialForwardingServerCallListener.onHalfClose(PartialForwardingServerCallListener.java:35)
    at io.grpc.ForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:23)
    at io.grpc.ForwardingServerCallListener$SimpleForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:40)
    at io.grpc.Contexts$ContextualizedServerCallListener.onHalfClose(Contexts.java:86)
    at io.grpc.PartialForwardingServerCallListener.onHalfClose(PartialForwardingServerCallListener.java:35)
    at io.grpc.ForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:23)
    at io.grpc.ForwardingServerCallListener$SimpleForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:40)
    at io.grpc.Contexts$ContextualizedServerCallListener.onHalfClose(Contexts.java:86)
    at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:331)
    at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:866)
    at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
    at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:750)

Mar 18, 2022 1:08:36 PM io.gapi.emulators.netty.HttpHandler$1 onError
INFO: Exception when handling request: UNKNOWN
*** shutting down gRPC server since JVM is shutting down
*** server shut down
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,3 +8,4 @@
 - Fixes issue where function deployment errored on projects without secrets (#4425).
 - Adds a blocking trigger type (#4395).
 - Removes verbose HTTP responses from debug logs (#4428)
+- Releases Cloud Firestore Emulator v1.14.3: fixes #4336.

--- a/src/emulator/downloadableEmulators.ts
+++ b/src/emulator/downloadableEmulators.ts
@@ -39,14 +39,14 @@ export const DownloadDetails: { [s in DownloadableEmulators]: EmulatorDownloadDe
     },
   },
   firestore: {
-    downloadPath: path.join(CACHE_DIR, "cloud-firestore-emulator-v1.14.2.jar"),
-    version: "1.14.2",
+    downloadPath: path.join(CACHE_DIR, "cloud-firestore-emulator-v1.14.3.jar"),
+    version: "1.14.3",
     opts: {
       cacheDir: CACHE_DIR,
       remoteUrl:
-        "https://storage.googleapis.com/firebase-preview-drop/emulator/cloud-firestore-emulator-v1.14.2.jar",
-      expectedSize: 60551603,
-      expectedChecksum: "0930bb09c080b52b1cbc70a91377ccd8",
+        "https://storage.googleapis.com/firebase-preview-drop/emulator/cloud-firestore-emulator-v1.14.3.jar",
+      expectedSize: 60442855,
+      expectedChecksum: "63517534875818689639ee5dee57dd52",
       namePrefix: "cloud-firestore-emulator",
     },
   },

